crates/edr_napi/src/context.rs

```diff
@@ -50,46 +49,17 @@ impl EdrContext {
 #[derive(Debug)]
 pub struct Context {
     pub state_root_generator: Arc<Mutex<RandomHashGenerator>>,
-    #[cfg(feature = "tracing")]
-    _tracing_write_guard: tracing_flame::FlushGuard<std::io::BufWriter<std::fs::File>>,
 }
 
 impl Context {
     /// Creates a new [`Context`] instance. Should only be called once!
     pub fn new() -> io::Result<Self> {
-        let fmt_layer = tracing_subscriber::fmt::layer()
-            .with_file(true)
-            .with_line_number(true)
-            .with_thread_ids(true)
-            .with_target(false)
-            .with_level(true)
-            .with_filter(EnvFilter::from_default_env());
-
-        let subscriber = Registry::default().with(fmt_layer);
-
-        #[cfg(feature = "tracing")]
-        let (flame_layer, guard) = {
-            let (flame_layer, guard) =
-                tracing_flame::FlameLayer::with_file("tracing.folded").unwrap();
-
-            let flame_layer = flame_layer.with_empty_samples(false);
-            (flame_layer, guard)
-        };
-
-        #[cfg(feature = "tracing")]
-        let subscriber = subscriber.with(flame_layer);
```

Comment:
I removed the flame layer, because we don't have a singleton object to store it on currently and I didn't manage to make it work with as a static variable tonight.

Reply:
If I'm not mistaken, this singleton still exists on the Hardhat side. We might just have to move its construction to `provider.ts`, alongside the `EdrProviderWrapper`.

In my work removing all component-based implementations this actually keeps existing, as enabling of tracing needs to happen in a binary (i.e. a shared library or executable). I'd like to avoid using `lazy_static` for this as construction/deconstruction ordering is not guaranteed for static variables. By having the context, it's possible for users to incorporate it in their memory management as it should be done safely.

Reply:
Ok makes sense. Are you ok with re-adding the flame tracer once the code structure on the JS side crystalized?